### ISSUE TITLE
Migrate kubenet bells and whistles to cni   

### ISSUE THREAD

**Issue by freehan** [kubernetes/kubernetes#38890]
Stuff to migrate:

1. sysctl bridge-nf-call-iptables
2. promiscuous bridge mode and ebtables deduplication
3. traffic shaper
4. hostport

**Comment by cmluciano** [comment]
@freehan Does that mean moving to https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/network/cni/cni.go ? I think I can accomplish 1) in tandem with #26005

**Comment by thockin** [comment]
@dnardo probably a dup of other(s)

**Comment by dnardo** [comment]
Although I cannot assign myself to this, I'm currently working on this.  #2 is pending merge in the CNI repo.

I have code ready for #1 and #3 (however we may need further discussion on #3 since it requires that the traffic shaper "knows" the pod IP on deletion, and right now we have no good way to do this in CNI)
#4 is being handled in another PR in the CNI repo.

I don't know of any dup issues, I'll take a look and I close any dups.

**Comment by cmluciano** [comment]
I'll assign this to myself tracking @dnardo. I think you send an email to the contrib-X SIG requesting member access to the kubernetes repo.

/assign

**Comment by thockin** [comment]
This seems unlikely for 1.7?

**Comment by freehan** [comment]
Revendoring the CNI plugins is pending on a new CNI release. 

Because CNI just went thru a separation between plugins and libcni, and there is no release tag in plugin repo.

**Comment by luxas** [comment]
@freehan Will that upcoming new (splitted-up) release be called v0.6.0?

**Comment by freehan** [comment]
@luxas I am not sure. 
@dcbw comment?

**Comment by feiskyer** [comment]
@freehan Seems there is still no clear plan of this? e.g. in which release kubenet will be deprecated?

**Comment by dnardo** [comment]
@feiskyer  There is a plan, some work has been done already to do this.  We will most likely target 1.9 for this since it's not super urgent to get in.

**Comment by luxas** [comment]
> We will most likely target 1.9 for this since it's not super urgent to get in.

Target what for v1.9? Deprecation? Removal? Feature parity?

**Comment by dnardo** [comment]
I've been targeting parity first.  Or we could decide to not bring in the bridging related workarounds to the CNI bridge plugin.  Some of these work-arounds are in already though.
Right now I'm waiting on https://github.com/containernetworking/plugins/pull/42 to be merged and then the last part would be getting the "tc" part of kubenet in CNI and that should be it for parity.
The next step I think would be to deprecate and then eventually remove it altogether. I would think we would want some mileage on the code though before we completely remove it.

**Comment by luxas** [comment]
@dnardo can you please write up a proposal document or google doc for this migration in order to make this roadmap transparent to the community?

**Comment by dnardo** [comment]
This was done sometime back  and discussed on the sig-network, here is a link, it's a little out of date but the general idea is there:

https://docs.google.com/document/d/1q2TxBYROCf3um0FS0PGAZdznbZGS-h1aIJZQAs9xu-A/edit

**Comment by luxas** [comment]
@dnardo that doc doesn't seem to be externally shared. Can you please share it with kubernetes-dev and/or the sig-network mailing lists?

**Comment by dnardo** [comment]
It's shared with kubernetes-sig-network@googlegroups.com  are you joined to that group?

**Comment by luxas** [comment]
yes. hah, strange, now works for me. not sure why it wouldn't let me in earlier...
Thanks, will look

**Comment by squeed** [comment]
Filed issue containernetworking/plugins#94 to implement a CNI traffic shaping plugin.

**Comment by m1093782566** [comment]
There is a proposal on [Support traffic shaping for CNI network plugin](https://github.com/kubernetes/community/pull/1893), which is intended to fix "traffic shaper", please check it out.

**Comment by jellonek** [comment]
so what is the status of this issue?

**Comment by luxas** [comment]
ping @kubernetes/sig-network-feature-requests

**Comment by cmluciano** [comment]
@caseydavenport @jingax10 @bowei I think we briefly discussed this at the Kubecon SIG Net meetup, but I do not recall what we determined were the missing features left.

**Comment by athenabot** [comment]
@m1093782566 @dnardo @jingax10
If this issue has been triaged, please comment `/remove-triage unresolved`.

If you aren't able to handle this issue, consider unassigning yourself and/or adding the `help-wanted` label.

🤖 I am a bot run by vllry. 👩‍🔬
